sale_cancel_reason/wizard/cancel_reason.py

```diff
@@ -0,0 +1,25 @@
+# © 2013 Guewen Baconnier, Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import fields, models
+
+
+class SaleOrderCancel(models.TransientModel):
+
+    """ Ask a reason for the sale order cancellation."""
+
+    _name = "sale.order.cancel"
+    _description = __doc__
+
+    reason_id = fields.Many2one("sale.order.cancel.reason", required=True)
+
+    def confirm_cancel(self):
+        self.ensure_one()
+        act_close = {"type": "ir.actions.act_window_close"}
+        sale_ids = self._context.get("active_ids")
```

Comment:
Any reason why private context is used?

Reply:
@DarkoNikolovski seems I miss this part. Thanks for pointing.